ast/compile_test.go

```diff
@@ -3341,6 +3341,376 @@ func TestCompilerMockVirtualDocumentPartially(t *testing.T) {
 	assertCompilerErrorStrings(t, c, []string{"rego_compile_error: with keyword cannot partially replace virtual document(s)"})
 }
 
+func TestCompilerCheckUnusedAssignedVar(t *testing.T) {
```

Comment:
it's not _glaring_, but perhaps we could have a nested closure case?
```
p { 
  x := 1
  { y | y := [ z | z:=[1,2,3][_]; z > 1 ][_] }
}
``` 
though I'm not sure right now how this would look after the previous compiler stages have processed it 🤔